draft-ietf-webtrans-http2.md

```diff
@@ -186,31 +197,57 @@ the site that requested the creation of the session.
 
 Upon receiving an extended CONNECT request with a `:protocol` field set to
 `webtransport`, the HTTP server checks if the identified resource supports
-WebTransport sessions. If the resource does not, the server SHOULD reply with status
-code 404 ({{Section 6.5.4 of !RFC7231}}). To accept a WebTransport session the
-server replies with 2xx status code. Before accepting a session, a server MUST
-ensure that it authorizes use of the session by the site identified in the
-`Origin` header.
+WebTransport sessions. If the resource does not, the server SHOULD reply with
+status code 404 ({{Section 6.5.4 of !RFC7231}}). If it does, it MAY accept the
+session by replying with a 2xx series status code, as defined in Section 15.3
+of {{!SEMANTICS=I-D.ietf-httpbis-semantics}}. The WebTransport server MUST
+verify the `Origin` header to ensure that the specified origin is allowed to
+access the server in question.
 
 From the client's perspective, a WebTransport session is established when the
 client receives a 200 response. From the server's perspective, a session is
 established once it sends a 200 response. Both endpoints MUST NOT send any
-WebTransport frames on a given session before that session is established.
-
-## Limiting the Number of Simultaneous Sessions
-
-From a flow control perspective, WebTransport sessions count against HTTP/2
-session flow control limits just like regular HTTP requests, since they are
-established via an HTTP CONNECT request. This document does not make any effort
-to introduce a separate flow control mechanism for WebTransport sessions. If
-the server needs to limit the rate of incoming requests, it has alternative
-mechanisms at its disposal:
-
-* `HTTP_STREAM_REFUSED` error code defined in {{!RFC7540}} indicates to the
-  receiving HTTP/2 stack that the request was not processed in any way.
-* HTTP status code 429 indicates that the request was rejected due to rate
-  limiting {{!RFC6585}}. Unlike the previous method, this signal is directly
-  propagated to the application.
+WebTransport capsules on a given session before that session is established.
+
+## Flow Control
+
+### Limiting the Number of Simultaneous Sessions
+
+This document defines a SETTINGS_MAX_WEBTRANSPORT_SESSIONS parameter that allows
+the server to limit the maximum number of concurrent WebTransport sessions on a
+single HTTP/3 connection.  The client MUST NOT open more sessions than
+indicated in the server SETTINGS parameters.  The server MUST NOT close the
+connection if the client opens sessions exceeding this limit, as the client and
+the server do not have a consistent view of how many sessions are open due to
+the asynchronous nature of the protocol; instead, it MUST reset all of the
```

Comment:
MUST reset is not good always. Maybe a 4xx (too many requests is 426?)

Reply:
Fixed in [167cad5](https://github.com/ietf-wg-webtrans/draft-webtransport-http2/pull/58/commits/167cad5b8fbcc1df3093bf2ec83fe598bdf6886c)